From 3405f9268c6cbdd75bdae4e060d267601d02580b Mon Sep 17 00:00:00 2001
From: Jonathan David <jonathan.david@ni.com>
Date: Tue, 15 Mar 2016 13:26:34 -0500
Subject: [PATCH] Add nicurl wrapper functions

Added the following API wrapper functions:
nicurl_easy_getinfo
nicurl_easy_setopt
nicurl_formadd
nicurl_multi_setopt
nicurl_share_setopt

These functions are called by older NI components, and exist as
a means to keep previous code running.

Upstream-Status: Inappropriate [NI specific changes]
---
 lib/easy.c     | 11 +++++++++++
 lib/formdata.c | 12 ++++++++++++
 lib/multi.c    | 11 +++++++++++
 lib/setopt.c   | 10 ++++++++++
 lib/share.c    | 11 +++++++++++
 5 files changed, 55 insertions(+)

diff --git a/lib/easy.c b/lib/easy.c
index 6b914353d..47b85f40d 100644
--- a/lib/easy.c
+++ b/lib/easy.c
@@ -857,6 +857,17 @@ CURLcode curl_easy_getinfo(struct Curl_easy *data, CURLINFO info, ...)
   return result;
 }
 
+#ifdef ENABLE_NICURL_API
+CURL_EXTERN CURLcode nicurl_easy_getinfo(CURL *curl, CURLINFO info, va_list arg)
+{
+/* Unlike other nicurl functions, this one did not previously call va_start()
+ * nor va_end(). Thus, it can behave as expected of variadic functions that
+ * are passed va_list.
+ */
+  return curl_easy_getinfo_va(curl, info, arg);
+}
+#endif
+
 static CURLcode dupset(struct Curl_easy *dst, struct Curl_easy *src)
 {
   CURLcode result = CURLE_OK;
diff --git a/lib/formdata.c b/lib/formdata.c
index f91241052..d73cd1fe2 100644
--- a/lib/formdata.c
+++ b/lib/formdata.c
@@ -704,6 +704,18 @@ CURLFORMcode curl_formadd(struct curl_httppost **httppost,
   return result;
 }
 
+#ifdef ENABLE_NICURL_API
+CURL_EXTERN CURLFORMcode nicurl_formadd(struct curl_httppost **httppost,
+                                        struct curl_httppost **last_post,
+                                        va_list arg)
+{
+  CURLFORMcode result;
+  result = curl_formadd_va(httppost, last_post, arg);
+  va_end(arg);
+  return result;
+}
+#endif
+
 /*
  * curl_formget()
  * Serialize a curl_httppost struct.
diff --git a/lib/multi.c b/lib/multi.c
index f85284695..abed7a4bb 100644
--- a/lib/multi.c
+++ b/lib/multi.c
@@ -2719,6 +2719,17 @@ CURLMcode curl_multi_setopt(struct Curl_multi *multi,
   return res;
 }
 
+#ifdef ENABLE_NICURL_API
+CURL_EXTERN CURLMcode nicurl_multi_setopt(CURLM *multi_handle,
+                            CURLMoption option, va_list param)
+{
+  CURLMcode result;
+  result = curl_multi_setopt_va(multi_handle, option, param);
+  va_end(param);
+  return result;
+}
+#endif
+
 /* we define curl_multi_socket() in the public multi.h header */
 #undef curl_multi_socket
 
diff --git a/lib/setopt.c b/lib/setopt.c
index af53ee3ef..84799a178 100644
--- a/lib/setopt.c
+++ b/lib/setopt.c
@@ -2595,3 +2595,13 @@ CURLcode curl_easy_setopt(struct Curl_easy *data, CURLoption tag, ...)
   va_end(arg);
   return result;
 }
+
+#ifdef ENABLE_NICURL_API
+CURL_EXTERN CURLcode nicurl_easy_setopt(CURL *curl, CURLoption tag, va_list arg)
+{
+  CURLcode result;
+  result = Curl_vsetopt(curl, tag, arg);
+  va_end(arg);
+  return result;
+}
+#endif
diff --git a/lib/share.c b/lib/share.c
index 870b191fc..109fcaecf 100644
--- a/lib/share.c
+++ b/lib/share.c
@@ -172,6 +172,17 @@ curl_share_setopt(struct Curl_share *share, CURLSHoption option, ...)
   return res;
 }
 
+#ifdef ENABLE_NICURL_API
+CURLSHcode
+CURL_EXTERN nicurl_share_setopt(CURLSH *sh, CURLSHoption option, va_list param)
+{
+  CURLSHcode result;
+  result = curl_share_setopt_va(sh, option, param);
+  va_end(param);
+  return result;
+}
+#endif
+
 CURLSHcode
 curl_share_cleanup(struct Curl_share *share)
 {
-- 
2.17.1

